Cython/Includes/cpython/dict.pxd

```diff
@@ -1,13 +1,40 @@
 from .object cimport PyObject
 
 cdef extern from "Python.h":
+    # On Python 2, PyDict_GetItemWithError is called _PyDict_GetItemWithError
+    """
+    #if PY_MAJOR_VERSION <= 2
+    #define PyDict_GetItemWithError _PyDict_GetItemWithError
+    #endif
+    """
+    # Macro defined in pyport.h
+    ctypedef unsigned long uint64_t
 
     ############################################################################
     # 7.4.1 Dictionary Objects
     ############################################################################
 
-    # PyDictObject
-    #
+    ctypedef struct PyDictEntry:
+        Py_ssize_t me_hash
+        PyObject* me_key
+        PyObject* me_value
+
+    ctypedef struct PyDictKeysObject
+
+    ctypedef class __builtin__.dict [object PyDictObject]:
+        cdef Py_ssize_t ma_used
+
+        # Older Python versions
+        cdef Py_ssize_t ma_fill
+        cdef Py_ssize_t ma_mask
+        cdef PyDictEntry* ma_table
+        cdef PyDictEntry* (*ma_lookup)(dict self, key, Py_hash_t hash) except NULL
+
+        # Newer Python versions
```

Comment:
That's probably Py3.6+, right? I think the exact version is worth mentioning here.